.github/workflows/wpcloud.yml

```
@@ -0,0 +1,150 @@
+        if: steps.changed.outputs.wpcomsh == 'true'
+        run: |
+          find . -path ./.github -prune -o -type f -print | sort > /tmp/before.txt
+          echo "::group::Installing and building wpcomsh"
+          pnpm jetpack build -v --deps plugins/wpcomsh
+          echo "::endgroup::"
+
+          # We only want to save the files that were actually created or changed.
+          # But we can't just list them for actions/cache/save, "Argument list too long".
+          # So instead we delete all the unchanged files so we can tell actions/cache/save
+          # to save everything that's left.
+          git -c core.quotepath=off diff --name-only | sort > /tmp/changed.txt
+          if [[ -s /tmp/changed.txt ]]; then
+            grep -F -x -v -f /tmp/changed.txt /tmp/before.txt > /tmp/remove.txt
+          else
+            cp /tmp/before.txt /tmp/remove.txt
+          fi
+          xargs -d '\n' rm < /tmp/remove.txt
+          find . -type d -empty -delete
+
+      - name: Save wpcomsh build cache
+        if: steps.changed.outputs.wpcomsh == 'true'
+        id: wpcomsh-build-cache-save
+        uses: actions/cache/save@v4
+        with:
+          path: |
+            .
+            !./.github/
+          key: ${{ github.sha }}
+  deploy:
+    name: Run PHPUnit on the WPCloud test site
+    runs-on: ubuntu-latest
+    needs: build
+    if: needs.build.outputs.wpcomsh == 'true'
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Restore wpcomsh build cache
+        id: wpcomsh-build-cache
+        uses: actions/cache/restore@v4
+        with:
+          path: |
+            .
+            !./.github/
+          key: ${{ github.sha }}
+          fail-on-cache-miss: true
+
+      - name: Setup tools
+        uses: ./.github/actions/tool-setup
+
+      - name: Install monorepo
+        run: |
+          pnpm install
+
+      - name: Configure Github to be able to SSH to the Atomic site
+        run: |
+          echo "::group::Intializing"
+
+          mkdir -vp ~/.ssh/
+          chmod -v 700 ~/.ssh
+
+          touch ~/.ssh/id_site
+          touch ~/.ssh/known_hosts
+          touch ~/.ssh/config
+          chmod 600 ~/.ssh/id_site
+          chmod 600 ~/.ssh/known_hosts
+          chmod 600 ~/.ssh/config
+          echo "$SSH_KEY" > ~/.ssh/id_site
+          echo "wrote ~/.ssh/id_site"
+          echo "$SSH_KNOWN_HOSTS" > ~/.ssh/known_hosts
+          echo "wrote ~/.ssh/known_hosts"
+          echo "Host jpwpcomsh" > ~/.ssh/config
+          echo "  Hostname sftp.wp.com" >> ~/.ssh/config
+          echo "  User wpcom-jetpackisbestpack-default-237778992" >> ~/.ssh/config
+          echo "  IdentityFile ~/.ssh/id_site" >> ~/.ssh/config
+          echo "  IdentitiesOnly yes" >> ~/.ssh/config
+
+          echo "::endgroup::"
+
+          echo "::group::Transferring wpcomsh to the testing server"
+          ssh jpwpcomsh "wp dereferenced freshen > /dev/null 2>&1"
+          ssh jpwpcomsh "rm -rf /tmp/old-* > /dev/null 2>&1"
```

Might be worth not throwing away the output, so if something fails we have a better chance of being able to quickly debug
```suggestion
          ssh jpwpcomsh "wp dereferenced freshen"
          ssh jpwpcomsh "rm -rf /tmp/old-*"
```

---

Let's leave it like this for now, so we don't clutter the CI output with setup noise.